lib/glob-helpers.cjs

```diff
@@ -110,21 +110,26 @@ function normalizeFileForMatching(cwd, file) {
 
 exports.normalizeFileForMatching = normalizeFileForMatching;
 
+const normalizeNegatedPattern = (cwd, pattern) => {
+	// Remove `!` from pattern
+	const patternPath = pattern.slice(1);
+	const relativePath = path.relative(cwd, patternPath);
+	return `!${relativePath}`;
+};
+
 function normalizePattern(pattern) {
+	const cwd = process.cwd();
```

Comment:
`cwd` needs to be an argument here. AVA determines the directory to which patterns are relevant and it may not be the CWD.

Reply:
> @asaid-0 it's been a while since I've had to think about glob patterns and file paths. Why does this solution do the trick?

@novemberborn Thank you for your feedback, actually, I used this solution assuming that we should also solve the issue with such patterns `tests//test1`, `tests/sub/../test2` not only the trailing slash `tests/test/`

if we need to just solve the trailing slash issue I wouldn't use `path.relative()`
what do you think?

Reply:
thank you for pointing this out, I see now I think we should receive the CWD as an argument in the `normalizeGlobs` here:
https://github.com/avajs/ava/blob/dc93f372e6a1af5f3ec4b90329d63e2f1e20e4a3/lib/globs.js#L35

Then we pass it to `normalizePatterns` call in the same function.

Also i noticed this line in cli.js
https://github.com/avajs/ava/blob/dc93f372e6a1af5f3ec4b90329d63e2f1e20e4a3/lib/cli.js#L396

which using nearly the same function path.relative, but path.resolve is being used in the second parameter to get the absolute path